pkg/asset/installconfig/aws/validation_test.go

```diff
@@ -168,6 +185,41 @@ func TestValidate(t *testing.T) {
 		}(),
 		availZones:     validAvailZones(),
 		privateSubnets: validPrivateSubnets(),
+	}, {
+		name: "valid instance types",
+		installConfig: func() *types.InstallConfig {
+			c := validInstallConfig()
+			c.Platform.AWS = &aws.Platform{Region: "us-east-1"}
+			c.ControlPlane.Platform.AWS.InstanceType = "m5.xlarge"
+			c.Compute[0].Platform.AWS.InstanceType = "m5.large"
+			return c
+		}(),
+		availZones:    validAvailZones(),
+		instanceTypes: validInstanceTypes(),
+	}, {
+		name: "invalid control plane instance type",
+		installConfig: func() *types.InstallConfig {
+			c := validInstallConfig()
+			c.Platform.AWS = &aws.Platform{Region: "us-east-1"}
+			c.ControlPlane.Platform.AWS.InstanceType = "t2.small"
+			c.Compute[0].Platform.AWS.InstanceType = "m5.large"
+			return c
+		}(),
+		availZones:    validAvailZones(),
+		instanceTypes: validInstanceTypes(),
+		expectErr:     `^\[controlPlane\.platform\.aws\.type: Invalid value: \"t2.small\": instance type does not meet minimum resource requirements of 4 vCPUs, controlPlane\.platform\.aws\.type: Invalid value: \"t2.small\": instance type does not meet minimum resource requirements of 16384 MiB Memory\]$`,
+	}, {
+		name: "invalid compute instance type",
+		installConfig: func() *types.InstallConfig {
+			c := validInstallConfig()
+			c.Platform.AWS = &aws.Platform{Region: "us-east-1"}
+			c.ControlPlane.Platform.AWS.InstanceType = "m5.xlarge"
+			c.Compute[0].Platform.AWS.InstanceType = "t2.small"
+			return c
+		}(),
+		availZones:    validAvailZones(),
+		instanceTypes: validInstanceTypes(),
+		expectErr:     `^\[compute\[0\]\.platform\.aws\.type: Invalid value: \"t2.small\": instance type does not meet minimum resource requirements of 2 vCPUs, compute\[0\]\.platform\.aws\.type: Invalid value: \"t2.small\": instance type does not meet minimum resource requirements of 8192 MiB Memory\]$`,
```

Comment:
would like to see one testcase for defaultMachinePool too.

Reply:
Added defaultMachinePool to the valid instance types test.